### PR TITLE
Embed Node.js Runtime #23

### PR DESCRIPTION
**Pull request by vrubezhny**
Fixes: #23

Signed-off-by: Victor Rubezhny <vrubezhny@redhat.com>

### CODE REVIEW

File: org.eclipse.wildwebdeveloper.embedder.node/src/org/eclipse/wildwebdeveloper/embedder/node/InitializeLaunchConfigurations.java [hunk 1]
```diff
@@ -14,21 +14,26 @@
  *   Pierre-Yves B. - Issue #238 Why does wildweb do "/bin/bash -c which node" ?
  *   Pierre-Yves B. - Issue #268 Incorrect default Node.js location for macOS
  *******************************************************************************/
-package org.eclipse.wildwebdeveloper;
+package org.eclipse.wildwebdeveloper.embedder.node;
```
Comment (mickaelistria) [inline org.eclipse.wildwebdeveloper.embedder.node/src/org/eclipse/wildwebdeveloper/embedder/node/InitializeLaunchConfigurations.java:17]: should we rename the class by the way, to something like `NodeJSManager` or `NodeJSAccessor` ?

Comment (vrubezhny) [reply org.eclipse.wildwebdeveloper.embedder.node/src/org/eclipse/wildwebdeveloper/embedder/node/InitializeLaunchConfigurations.java:17]: I like `NodeJSManager` or `NodeJSRuntimeProvider`. But the first name is better as it's shorter.

File: org.eclipse.wildwebdeveloper.embedder.node/src/org/eclipse/wildwebdeveloper/embedder/node/InitializeLaunchConfigurations.java [hunk 2]
```diff
@@ -70,13 +102,12 @@ public static String getNodeJsLocation() {
 		return null;
 	}
 
-	public static String which(String program) {
-
+	public static File which(String program) {
```
Comment (vrubezhny) [inline org.eclipse.wildwebdeveloper.embedder.node/src/org/eclipse/wildwebdeveloper/embedder/node/InitializeLaunchConfigurations.java:105]: Should we make `which()` to search for the specified program in the embedded NodeJS executable installation folder first, then using other possible paths?

**Comment by vrubezhny** [comment]
Last changes:
- Linux and MacOS fragments are switched to use *.tar.gz formats of NodeJS distribution as it gives x2.6 times faster decompression even given bigger size of the distribution archives. 
- InitializeLaunchConfigurations renamed to NodeJSManager;
- which() is made to search first inside embedded NodeJS installation folder;
- Node Debug is configured to use embedded NodeJS (if embedding takes place) instead of any occasionally installed NodeJS distribution.

**Comment by mickaelistria** [comment]
All that seems really good!
I'm merging it so we can more easily test. We need to test it on environment where node.js is absent. I'll probably tweak the Jenkinsfile later to have tests running without node.js in path.

**Pull request merged**